query-parser/src/parse.rs

```diff
@@ -65,6 +67,11 @@ error_chain! {
     }
 
     errors {
+        DuplicateVariableError {
```

Comment:
Unfortunately it's a little tricky with the way I've implemented it; I hope this can be part of an ergonomics pass in a few months. It should be pretty obvious to users when there are duplicated variables.